src/math/hypot.rs

```diff
@@ -0,0 +1,74 @@
+use core::f64;
+
+use super::sqrt;
+
+const SPLIT: f64 = 134217728. + 1.; // 0x1p27 + 1 === (2 ^ 27) + 1
```

Comment:
I'm not sure if this is correct; the source is:
```c
#if FLT_EVAL_METHOD > 1U && LDBL_MANT_DIG == 64
#define SPLIT (0x1p32 + 1)
#else
#define SPLIT (0x1p27 + 1)
#endif
```
I don't fully understand `FLT_EVAL_METHOD`, but I believe that Rust semantics are equivalent to `#define FLT_EVAL_METHOD 0`

Reply:
Both `FLT_EVAL_METHOD` and `FLT_EVAL_METHOD` are defined in the MUSL source code (`grep -r 'define.*FLT_EVAL_METHOD`) and their value depends on the target architecture. Though it seems that for all the architectures that Rust supports the if condition will eval to false so this is good as it is.

If that were not the case this would need cfg attributes of this sort: `#[cfg(any(target_arch = "x86_64", target_arch = "aarch64")]`.